xr_sample_project/Assets/SampleViewer/Samples/VRSample/Scripts/UITextSlider.cs

```diff
@@ -0,0 +1,89 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+public class UITextSlider : MonoBehaviour
+{
+    [SerializeField] private TextMeshProUGUI sliderText;
+    [SerializeField] private Slider slider;
+    [SerializeField] private int continuousMovementValue = -1;
+
+    [SerializeField] private SetTurnType turnTypeScript;
+    CanvasGroup canvasGroup;
+    // Start is called before the first frame update
+    void Start()
+    {
+        slider.onValueChanged.AddListener(ChangeSliderValue);
+
+        canvasGroup = GetComponent<CanvasGroup>();
+
+        sliderText.text = slider.value.ToString();
+    }
+
+    private void OnEnable()
+    {
+        if (continuousMovementValue > 1)
+        {
+            turnTypeScript.OnTypeChanged += EnableSlider;
+        }
+    }
+
+    private void OnDisable()
+    {
+        if (continuousMovementValue > 1)
+        {
+            turnTypeScript.OnTypeChanged -= EnableSlider;
+        }
+    }
+
+    void ChangeSliderValue(float value)
+    {
+        sliderText.text = value.ToString();
+        if(continuousMovementValue >= 0)
+        {
+            ContinuousMovement continuousMovement = FindAnyObjectByType<ContinuousMovement>();
+            SetTurnType turnTypeScript = FindAnyObjectByType<SetTurnType>();
+            
+            if(continuousMovement != null )
+            {
+                switch (continuousMovementValue)
+                {
+                    case 0:
+                        continuousMovement.SetSpeed(value);
```

Comment:
```suggestion
                        continuousMovement?.SetSpeed(value);
```
Do this for the set speed calls and you wont need the null check on 48. Or return early if null.